packages/web-components/src/utils/helpers.ts

```diff
@@ -419,7 +419,7 @@ export const hasValidationStatus = (
 };
 
 export const isSlotUsed = (element: HTMLElement, slotName: string): boolean => {
-  return !!element.querySelector(`[slot="${slotName}"]`);
+  return Array.from(element.children).some(child => child.getAttribute('slot') === slotName);
```

Comment:
Might be worth adding a check that the elements.children is HTMLCollection[] so we do not get undefined or null being passed into Array.from()
```javascript
  return HTMLCollection.prototype.isPrototypeOf(element.children)
    ? Array.from(element.children).some(el => el.getAttribute('slot') === slotName)
    : false;
```

Reply:
As far as I can tell this check shouldn't be necessary as the helper function handles null/undefined by creating an empty array if the element has no children.